File: p_42577_phonnum/phonnum.cpp
#include<iostream>
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#define debug
#define FILENAME "phonnum_in.txt"
#define FILE_CIN
using namespace std;
bool solution(vector<string> phoneBook) {
    bool answer = true;
    sort(phoneBook.begin(), phoneBook.end());
    cout<<phoneBook[0]<<endl;
    for ( int i = 0 ; i < phoneBook.size() - 1 ; i++ )
    {
        string we = phoneBook[i+1].substr(0,phoneBook[i].size());
        if ( phoneBook[i] ==  we)
        {
            cout <<phoneBook[i]<<' '<<phoneBook[i+1]<<endl;
            answer = false;
            break;
        }
    }

    return answer;
}

int main(){
#ifdef FILE_CIN
	freopen(FILENAME,"r",stdin);
	int tmp;
	cin>>tmp;
	for(int z=0;z<tmp;z++){
#endif
#ifdef FILE_CIN
	}
#endif
	return 0;
}

File: p_42577_phonnum/phonnum.py
def solution(phone_book):
    """
    answer = True
    for s in phone_book:
        tmp=""
        for letter in s:
            tmp+=letter
            if tmp in phone_book:
                if tmp != s:
                    answer=False
    return answer
    """
    answer = True
    di={}
    for s in phone_book:
        di[s]=1
    for s in phone_book:
        tmp=""
        for letter in s:
            tmp+=letter
            if tmp in di:
                if tmp != s:
                    answer=False
    return answer
